### PR TITLE
GitHub Actions (macOS)

### PR DESCRIPTION
**Pull request by mrexodia**
This GitHub Actions script is easily extensible to include other operating systems and/or compilers. But it might be a bit difficult to get the tests running on all platforms properly.

Related: #133

### CODE REVIEW

**Comment by mrexodia** [comment]
@c3rb3ru5d3d53c Maybe you could take a look at setting up the tests? After the cmake install of pybinlex I was able to `import pybinlex` from the build/install/bin folder.

The Makefile is black magic to me, maybe some python script to handle the unpacking of the test binaries and executing all the tests would be an idea? That should work on all platforms and then the GitHub Actions script can be uniform on all operating systems.

**Comment by c3rb3ru5d3d53c** [comment]
@mrexodia Yeah we can look at doing a script so it works on all platforms, `Python` is universal to all of them so that can easily be done.

**Comment by c3rb3ru5d3d53c** [comment]
We also should have actions not just for macos but also actions for Linux and Windows too

File: .github/workflows/cmake.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+jobs:
+  build:
+    # Skip building pull requests from the same repository
+    if: ${{ github.event_name == 'push' || (github.event_name == 'pull_request' && github.event.pull_request.head.repo.full_name != github.repository) }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      # TODO: replace extra-cmake-flags with CMake presets
+      matrix:
+        target:
```
Comment (mrexodia) [inline .github/workflows/cmake.yml:14]: @c3rb3ru5d3d53c if you want to enable other platforms you can add an entry to the `target` list and the corresponding variables in `include`. Example I did yesterday here: https://github.com/li-script/lightning/blob/master/.github/workflows/cmake.yml#L13

Comment (c3rb3ru5d3d53c) [reply .github/workflows/cmake.yml:14]: Perfect I'll look into that soon enough :)

**Comment by mrexodia** [comment]
Was my matrix strategy not working?

**Comment by c3rb3ru5d3d53c** [comment]
So getting some errors, likely due to matrix environment vars not being passed correctly.

```
-- Build files have been written to: /home/runner/work/binlex/binlex/build
CMake Error: Invalid value used with --config
```

Not sure as to why, will have to investigate further

**Comment by c3rb3ru5d3d53c** [comment]
Matrix is working just be being dumb lol, not much sleep lately :)

**Pull request merged**